processor/src/advice/mod.rs

```diff
@@ -19,47 +70,95 @@ use vm_core::{
 ///    trees and can be used to provide Merkle paths.
 ///
 /// An advice provider can be instantiated from [ProgramInputs].
-pub struct AdviceProvider {
+#[derive(Debug, Clone, Default)]
+pub struct BaseAdviceProvider {
     step: u32,
     tape: Vec<Felt>,
     values: BTreeMap<[u8; 32], Vec<Felt>>,
     sets: BTreeMap<[u8; 32], AdviceSet>,
```

Comment:
I don't love these names. I wonder if we can come up with something better. `values` is really a key-value map, and `sets` is all about Merkle paths. We don't need to do anything about this in this PR - but something to think about in the future.

Reply:
They could be `assist_tapes` and `merkle_sets`, respectively. wdyt?

Reply:
I like `merkle_sets`! `assist_tapes` to me still doesn't capture the meaning. I'll try to think more about it too.